disnake/guild.py

```diff
@@ -4553,3 +4566,468 @@ async def create_automod_rule(
+                        "id": target,
+                        # can only set overrides for roles here
+                        "type": abc._Overwrites.ROLE,
+                    }
+                )
+            data["permission_overwrites"] = overwrites_data
+
+        if category is not MISSING:
+            data["parent_id"] = category
+
+        if topic is not MISSING:
+            data["topic"] = topic
+
+        if slowmode_delay is not MISSING:
+            data["rate_limit_per_user"] = slowmode_delay
+
+        if nsfw is not MISSING:
+            data["nsfw"] = nsfw
+
+        self._channels.append(data)
+        return _id, data
+
+    def add_category(
+        self,
+        name: str,
+        *,
+        overwrites: Dict[PlaceholderID, PermissionOverwrite] = MISSING,
+    ) -> PlaceholderID:
+        """
+        Adds a category channel to the guild builder.
+
+        There is an alias for this named ``add_category_channel``.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The category's name.
+        overwrites: Dict[``PlaceholderID``, :class:`PermissionOverwrite`]
+            A :class:`dict` of roles to :class:`PermissionOverwrite`\\s which can be synced to channels.
+
+        Returns
+        -------
+        ``PlaceholderID``
+            A placeholder ID for the created category.
+        """
+        _id, _ = self._add_channel(type=ChannelType.category, name=name, overwrites=overwrites)
+        return _id
+
+    add_category_channel = add_category
+
+    def add_text_channel(
+        self,
+        name: str,
+        *,
+        overwrites: Dict[PlaceholderID, PermissionOverwrite] = MISSING,
+        category: PlaceholderID = MISSING,
+        topic: Optional[str] = MISSING,
+        slowmode_delay: int = MISSING,
+        nsfw: bool = MISSING,
+        default_auto_archive_duration: AnyThreadArchiveDuration = MISSING,
+    ) -> PlaceholderID:
+        """
+        Adds a text channel to the guild builder.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The channel's name.
+        overwrites: Dict[``PlaceholderID``, :class:`PermissionOverwrite`]
+            A :class:`dict` of roles to :class:`PermissionOverwrite`\\s to apply to the channel.
+        category: ``PlaceholderID``
+            The category to place the new channel under.
+
+            .. warning::
+                Unlike :func:`Guild.create_text_channel`, the parent category's
+                permissions will *not* be synced to this new channel by default.
+
+        topic: Optional[:class:`str`]
+            The channel's topic.
+        slowmode_delay: :class:`int`
+            Specifies the slowmode rate limit for users in this channel, in seconds.
+            A value of ``0`` disables slowmode. The maximum value possible is ``21600``.
+            If not provided, slowmode is disabled.
+        nsfw: :class:`bool`
+            Whether to mark the channel as NSFW or not.
```

Comment:
This is optional, but in #275, I changed all "Whether ......." documentation and leave it without "or not"/"not" (there are still some documentation with that), but it'd be nice to remove "or not".

(same below)

Reply:
https://github.com/DisnakeDev/disnake/pull/578/commits/fb57ad630e91953b2dc693cf0aa512abc725be77